From 421070c3a9d69ad463cfdafd75b22e75d1a55b7e Mon Sep 17 00:00:00 2001
From: Lin Ruizhe <linruizhe@huawei.com>
Date: Fri, 31 Dec 2021 15:59:22 +0800
Subject: [PATCH] interrupt: Add kabi_reserve in interrupt.h
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 421070c3a9d69ad463cfdafd75b22e75d1a55b7e


hulk inclusion
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MZU1
CVE: NA

---------------------------

Add kabi_reserve tasklet_struct and irq_affinity in
interrupt.h

Signed-off-by: Lin Ruizhe <linruizhe@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/interrupt.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/interrupt.h b/include/linux/interrupt.h
index 9e57d44d98c5..22240a8c3a1e 100644
--- a/include/linux/interrupt.h
+++ b/include/linux/interrupt.h
@@ -13,6 +13,7 @@
 #include <linux/hrtimer.h>
 #include <linux/kref.h>
 #include <linux/workqueue.h>
+#include <linux/kabi.h>
 
 #include <linux/atomic.h>
 #include <asm/ptrace.h>
@@ -302,6 +303,7 @@ struct irq_affinity {
 	unsigned int	set_size[IRQ_AFFINITY_MAX_SETS];
 	void		(*calc_sets)(struct irq_affinity *, unsigned int nvecs);
 	void		*priv;
+	KABI_RESERVE(1)
 };
 
 /**
@@ -597,6 +599,7 @@ struct tasklet_struct
 		void (*callback)(struct tasklet_struct *t);
 	};
 	unsigned long data;
+	KABI_RESERVE(1)
 };
 
 #define DECLARE_TASKLET(name, _callback)		\
-- 
2.26.2

